src/Optimizer/Transformer/AmpStoryCssOptimizer.php

```diff
@@ -0,0 +1,244 @@
+    {
+        $ampStoryCssLink = $document->createElementWithAttributes(Tag::LINK, [
+            Attribute::REL           => Attribute::REL_STYLESHEET,
+            Attribute::AMP_EXTENSION => Extension::STORY,
+            Attribute::HREF          => Amp::CACHE_HOST . '/v0/amp-story-1.0.css',
+        ]);
+
+        $document->head->appendChild($ampStoryCssLink);
+    }
+
+    /**
+     * Replace viewport units in custom css with related css variables.
+     *
+     * @param Element $style The style element to modify.
+     */
+    private function modifyAmpCustomCSS(Element $style)
+    {
+        $style->nodeValue = preg_replace(
+            '/(-?[\d.]+)v(w|h|min|max)/',
+            'calc($1 * var(--story-page-v$2))',
+            $style->nodeValue
+        );
+    }
+
+    /**
+     * Append an inline script tag for the dvh polyfill
+     *
+     * @param Document $document The document in which we need to append the script tag.
+     * @return void
+     */
+    private function appendAmpStoryDvhPolyfillScript(Document $document)
+    {
+        $ampStoryDvhPolyfillScript = $document->createElementWithAttributes(Tag::SCRIPT, [
+            Attribute::AMP_STORY_DVH_POLLYFILL => '',
+        ], self::AMP_STORY_DVH_POLYFILL_CONTENT);
+
+        $document->head->appendChild($ampStoryDvhPolyfillScript);
+    }
+
+    /**
+     * Add data-story-supports-landscape attribute to support landscape.
+     *
+     * @param Document $document The document in which we need to add the attribute.
+     */
+    private function supportsLandscapeSSR(Document $document)
+    {
+        /**
+         * @var Element|null
+         */
+        $story = $document->body->getElementsByTagName(Extension::STORY)->item(0);
+
+        if (! $story) {
+            return;
+        }
+
+        if ($story->hasAttribute(Attribute::SUPPORTS_LANDSCAPE)) {
+            $document->html->setAttribute(Attribute::DATA_STORY_SUPPORTS_LANDSCAPE, '');
+        }
+    }
+
+    /**
+     * Add aspect-ratio inline style for amp-story-grid-layer.
+     *
+     * @param Document $document The document in which we need to add the style.
+     */
+    private function aspectRatioSSR(Document $document)
+    {
+        for ($node = $document->body; $node !== null; $node = NodeWalker::nextNode($node)) {
+            if (! $node instanceof Element) {
+                continue;
+            }
+
+            if (Amp::isTemplate($node)) {
+                $node = NodeWalker::skipNodeAndChildren($node);
+                continue;
+            }
+
+            if ($node->tagName !== Extension::STORY_GRID_LAYER) {
+                continue;
+            }
+
+            if (! $node->hasAttribute(Attribute::ASPECT_RATIO)) {
+                continue;
+            }
+
+            $aspectRatio = str_replace(':', '/', $node->getAttribute(Attribute::ASPECT_RATIO));
+
+            $node->addInlineStyle('--' . Attribute::ASPECT_RATIO . ':' . $aspectRatio, true);
```

Comment:
This is technically not the aspect ratio attribute (for which the constant is being used), but the AMP Story aspect ratio CSS variable.
```suggestion
            $node->addInlineStyle("--aspect-ratio:{$aspectRatio}", true);
```